src/helper/streaming/index.ts

```diff
@@ -1,59 +1,21 @@
-import type { Context } from '../../context'
+import type { Context, HeaderRecord } from '../../context'
+import type { StatusCode } from '../../utils/http-status'
 import { StreamingApi } from '../../utils/stream'
 
-interface SSEMessage {
-  data: string
-  event?: string
-  id?: string
+export const stream = (
```

Comment:
We can make it simpler:

```ts
export const stream = (
  c: Context,
  cb: (stream: StreamingApi) => Promise<void>,
): Response => {
  const { readable, writable } = new TransformStream()
  const stream = new StreamingApi(writable)
  cb(stream).finally(() => stream.close())
  return c.newResponse(readable)
}
```

Users can set the status and the headers with `c.status()` and `c.header(),` so I think it does not need `arg?` and `headers?`. `steamSSE()` also follows this approach.